server/controllers/cashier.js

```diff
@@ -0,0 +1,78 @@
+import '@babel/plugin-transform-regenerator';
+import '@babel/polyfill';
+import pool from '../config/db-config';
+
+class Cashier {
+  static async debitAccount(req, res) {
+    const cashierId = req.userData.id;
+
+    const getCashier = `SELECT * FROM users WHERE id= '${cashierId}';`;
+    const { rows: [cashierGet] } = await pool.query(getCashier);
+   
+
+
+    const transaction = {
+      date: req.body.date,
+      type: 'debit',
+      accountnNumber: req.params.accountNo,
+      cashier: cashierGet.id,
+      amount: parseFloat(req.body.amount),
+    };
+
+
+    const accountGet = `SELECT * FROM accounts WHERE accountno = '${req.params.accountNo}';`;
+    const { rows: [rows2] } = await pool.query(accountGet);
+    if (!rows2) {
+      return res.status(404).json({
+        status: 404,
+        error: 'account number not found',
+      });
+    }
+
+
+    if (rows2.status === 'DRAFT' || rows2.status === 'dormant') {
+      return res.status(400).json({
+        status: 400,
+        error: 'account must be ACTIVATED to be used.',
+      });
+    }
+
+    const oldBalance = rows2.balance;
+
+    if (oldBalance === 0 || oldBalance <= transaction.amount) {
+
+      return res.status(400).json({
+        status: 400,
+        error: 'INSUFFICIENT BALANCE ON THE ACCOUNT',
+      });
+    }
+    const newBalance = oldBalance - transaction.amount;
```

Comment:
Good job, @irfiacre there's one more condition you need to handle. You need to make sure the transaction amount inputted is positive and not negative. Reach out for clarification if this isn't clear.

Reply:
Should I convert the amount(negative) to a positive number or just return an error?

Reply:
Return an error.